### DISCUSSION TITLE
Github projectsv2 importing question (enterprise)

### DISCUSSION THREAD

**Discussion by arnabkd** [community/community#67963] [Enterprise]
### Select Topic Area

Question

### Feature Area

Projects

#### Background
We are looking to migrate our task board from Trello to github projectsV2, and since we have a lot of cards (around 5000), we want to do it programmatically. 

#### Plan for exporting to Github
With our list of archived cards from Trello, we want to do the following:
- We have a repository called "archived-cards-from-trello" where we want to create issues for each historical Trello card
- We want to close *all* of these issues
- We want the project board to have all of these issues in the archive.

#### Problems we ran into
First of all, when we try to create the issues in the repository and link them automatically to the project, we get an error:
```
mutation createIssue {
  createIssue(input: {  projectIds: ["PROJECT_ID"] , repositoryId: "<REPO_ID>", title: "test creating issue linked to project board", body: "test body"}){
    issue {
      id
      url
    }
  }
}
```

The error is as follows:
```
{
  "data": {
    "createIssue": null
  },
  "errors": [
    {
      "type": "NOT_FOUND",
      "path": [
        "createIssue"
      ],
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "message": "Could not resolve to a node with the global id of ''<PROJECT_ID>"
    }
  ]
}
```

Now, I know that the project ID is correct, because when I got it from doing this query where I also confirmed that the repository in question was also linked to the project:
```
{
  viewer {
    organization(login: "<ORG_LOGIN>") {
      projectV2(number: <PROJECT_NUMBER>) {
        projectId: id 
        viewerCanUpdate
        title
        repositories(first: 1) {
          nodes {
            id
            name
          }
        }
      }
    }
  }
}
```

What am I doing wrong here? Is it simply because the `createIssue` mutation takes in `projectIds` that are the classic project IDs and this leads to an error when I supply the id of a project that is a V2? If that is indeed the case, should we make this using the "old" projects endpoint?

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by rileybroughten** [comment]
👋 You are correct that `createIssue` mutation only accepts `projectIds` of Projects (classic). I would suggest using [addProjectV2ItemById](https://docs.github.com/en/graphql/reference/mutations#addprojectv2itembyid) instead to add items to a project.

Let me know if this helps or if you have any additional questions!
